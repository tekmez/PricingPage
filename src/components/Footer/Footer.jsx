import React from 'react'
import { Text, View } from 'react-native'
import styles from './Footer.style'
import Link from '../Link'
const Footer = ({ data }) => {
  return (
    <View testID="footer-container" style={styles.container}>
      {data?.map((link, index) => (
        <React.Fragment key={index}>
          <Link url={link.url} title={link.title} />
          {index < data.length - 1 && <Text style={styles.link}> - </Text>}
        </React.Fragment>
      ))}
    </View>
  )
}

export default Footer
