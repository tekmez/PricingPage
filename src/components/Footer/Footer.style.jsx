import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    gap: 8,
  },
  link: {
    fontSize: 12,
    lineHeight: 16,
    color: '#717578',
    fontFamily: 'Inter-Medium',
  },
})

export default styles
