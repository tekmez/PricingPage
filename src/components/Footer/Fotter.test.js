import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import Footer from './Footer'

const footerLinks = [
  { title: 'Privacy Policy', url: 'https://www.google.com' },
  { title: 'Terms of Service', url: 'https://www.google.com' },
  { title: 'Contact Us', url: 'https://www.google.com' },
]
describe('Footer component', () => {
  it('should default match snapshot', () => {
    const comp = render(<Footer />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('should match snapshot with data', () => {
    const comp = render(<Footer data={footerLinks} />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('renders correctly and navigates to correct URL when pressed', () => {
    const comp = render(<Footer data={footerLinks} />)
    footerLinks.forEach((link) => {
      const linkElement = comp.getByText(link.title)
      fireEvent.press(linkElement)
      expect(linkElement).toBeTruthy()
    })
  })
})
