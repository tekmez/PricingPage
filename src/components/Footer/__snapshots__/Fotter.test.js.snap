// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should default match snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 8,
      "justifyContent": "center",
    }
  }
  testID="footer-container"
/>
`;

exports[`Footer component should match snapshot with data 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 8,
      "justifyContent": "center",
    }
  }
  testID="footer-container"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="link"
  >
    <Text
      style={
        {
          "color": "#717578",
          "fontFamily": "Inter-Medium",
          "fontSize": 12,
          "lineHeight": 16,
        }
      }
      testID="link-title"
    >
      Privacy Policy
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#717578",
        "fontFamily": "Inter-Medium",
        "fontSize": 12,
        "lineHeight": 16,
      }
    }
  >
     - 
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="link"
  >
    <Text
      style={
        {
          "color": "#717578",
          "fontFamily": "Inter-Medium",
          "fontSize": 12,
          "lineHeight": 16,
        }
      }
      testID="link-title"
    >
      Terms of Service
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#717578",
        "fontFamily": "Inter-Medium",
        "fontSize": 12,
        "lineHeight": 16,
      }
    }
  >
     - 
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="link"
  >
    <Text
      style={
        {
          "color": "#717578",
          "fontFamily": "Inter-Medium",
          "fontSize": 12,
          "lineHeight": 16,
        }
      }
      testID="link-title"
    >
      Contact Us
    </Text>
  </View>
</View>
`;
